src/main/jit/Optimization.cpp

```diff
@@ -53,8 +85,8 @@ char BasicFunctionPass::pass_id = 0;
 //------------------------------------------------------------------------------
 // Impelementation of RemoveRedundantCallsToSetVisibility.
 
-bool RemoveRedundantCallsToSetVisibility::runOnBasicBlock(
-    llvm::BasicBlock& block) {
+int RemoveRedundantCallsToSetVisibility::RemoveRedundantCalls(
+    llvm::BasicBlock& block, bool leave_one) {
```

Comment:
Unfortunately, eraseFromParent() doesn't seem to work with reverse iterators.  But, it might work with forward iterators. Let me see if I can simplify the code using that.